### ISSUE TITLE
Let players leave a group at will

### ISSUE THREAD

**Issue by StephanBijzitter** [wise-old-man/wise-old-man#1559]
I've got several clan members who still appear in the groups of their old clans, either because those clans have not updated their memberlists or because they use the option to add members rather than override members.

Now I'm not sure how this can be done, but a few ideas:

A player could go to their profile, visit the groups tab and for each group have a button to request their removal. The owner of the group will then be able to accept/reject the request in a new tab on their group's page. The request could be automatically accepted after X days of inaction.

A player could go to their profile, visit the groups tab and for each group have a button to request their removal. Upon requesting removal, the page tells them to perform some very specific actions in-game that results in specific XP gains in specific skills. If done precisely and within X amount of time, the player's identify (or control over the account) is confirmed and the request for removal is approved by the system.

**Comment by StephanBijzitter** [comment]
Oh and I'm pretty sure we can be seen together on this screenshot I made of a video I remembered:
![image](https://github.com/user-attachments/assets/0a89924a-1668-4991-b12b-bfbaa80bbde7)

**Comment by psikoi** [comment]
> Oh and I'm pretty sure we can be seen together on this screenshot I made of a video I remembered: 

Yes! I'm pretty sure that's me on the far left!

**Comment by psikoi** [comment]
As for the suggestion:

Unfortunately since we don't have user accounts (and most importantly, no way to verify if a user account has access to any given runescape account), anyone could click that "request to leave" button for everyone else

This might be fine, but also could become problematic in terms of spam

I can imagine a disgruntled ex-member of a clan spamming those requests just to annoy their ex-clan leader

We're very limited on features we can build because of the lack of runescape acc verification, but I'm happy to keep discussing

**Comment by StephanBijzitter** [comment]
Yeah, though I do think something along these lines would work:

* unverified player requests removal from a group
* system takes snapshot of hiscores
* website tells users to perform some specific actions
  * Cook 3 sharks for 330 XP
  * and Fletch 10 maple longbow (u)s for 583 XP
  * and Runecraft body runes from 4 rune essence for 30 XP
  * then logout
* unverified player has performed the actions and clicks on new button to verify
* system takes new snapshot of hiscores
* system compares before and after snapshots to confirm the actions have been performed on the player account in question and removes the player from the group (or not).

The chance of any player gaining these exact amounts of xp in these exact skills, within (let's say) an hour I would estimate to be so close to zero that an assumption of zero could be made.

Obviously the list could include any skill (aside from a few perhaps, like Defence) and many more possible XP values, this is just an example.

Does seem like a decent amount of work though, but at least it could be fully automated and without requiring any sort of actual registration.

**Comment by psikoi** [comment]
Yep! You nailed it

That was one of our potential verification strategies, we call them exp challenges. This could definitely work, but it would be a bit wonky for users to understand and execute + a lot more work on our end (as opposed to some web service Jagex could provide)

Some alternatives we thought of were:

- Forum-based authentication (where players would post a 6 digit code to a thread on the forums)
  - This is now impossible as the forums have been removed 😭
- Some sort of RuneLite based authentication where someone would say something like #wom-84923 and 20 other untrusted clients would report seeing this message to our server
  - This doesn't work at the plugin's current user base, unless participants are in a CC or something

And to state the obvious, yes, an ingame bot would definitely solve all these problems as players could just DM the bot a 6 digit code, that would be 100000% easier than anything else

**Comment by psikoi** [comment]
Oh to add some context, the exp challenges idea would rely on quite a bit of RL plugin work, and it's not super easy to find people willing to help write Java lmao, so the idea never got far

**Comment by StephanBijzitter** [comment]
Yeah the forum would have been amazing, rip.

RuneLite plugin (even with a high-enough userbase) would be open to abuse though, as all plugins are open-source someone with a bunch of proxies available could easily fake the required traffic. And even if the plugins were not open-source, WireShark is a thing.

And yeah a bot would be incredibly easy, though that would definitely require permission from Jagex to make sure the account doesn't get banned for staying online 24-7. They probably won't want to open than can of worms.

But... you did mention the plugin, which made me look at that repository and saw your list which includes https://github.com/wise-old-man/wiseoldman-runelite-plugin/issues/2 
If that message could mention removing members that have left, that may very well solve the issue for some of the players (in my clan) that are still listed as part of other clans.

And I do have myself some experience with RuneScape and Java, cough cough. I'd be happy to help out and tackle a bunch of things on your list.

**Comment by psikoi** [comment]
> And I do have myself some experience with RuneScape and Java, cough cough. I'd be happy to help out and tackle a bunch of things on your list.

Yeah that'd be great, come and chat with us on discord at https://wiseoldman.net/discord
